packages/flutter_tools/lib/src/dart_pub_json_formatter.dart

```diff
@@ -16,31 +16,24 @@ class DartDependencyPackage {
   });
 
   factory DartDependencyPackage.fromHashMap(dynamic packageInfo) {
-    String name = '';
-    String version = '';
-    String source = '';
-    List<dynamic> dependencies = <dynamic>[];
+    List<String> dependencies = <String>[];
 
-    if (packageInfo is LinkedHashMap) {
-      final LinkedHashMap<String, dynamic> info = packageInfo as LinkedHashMap<String, dynamic>;
-      if (info.containsKey('name')) {
-        name = info['name'] as String;
-      }
-      if (info.containsKey('version')) {
-        version = info['version'] as String;
-      }
-      if (info.containsKey('source')) {
-        source = info['source'] as String;
-      }
-      if (info.containsKey('dependencies')) {
-        dependencies = info['dependencies'] as List<dynamic>;
-      }
+    if (packageInfo is! LinkedHashMap) {
+      return DartDependencyPackage(
+        name: '',
+        version: '',
+        source: '',
+        dependencies: dependencies,
+      );
+    }
+    if (packageInfo case {'dependencies': final List<dynamic> list}) {
```

Comment:
Nit: can this be `List<Object?>` instead? We should avoid using `dynamic` whenever possible.